### ISSUE TITLE
Deeplink Tool returnURL

### ISSUE THREAD

**Issue by mtravis-interpretive** [celtic-project/LTI-PHP#89]
After our recent upgrade, we found out that our deep linking logic broke because the returnUrl is no longer being set.  We could manually pull that from the messageParameters, but you were parsing it into an easier to validate variable for us before.

In Tool->getMessageParameters, the code used to look like this so it would parse the messageParameters if they weren't set and parse the returnUrl in the same step:

```
public function getMessageParameters(bool $strictMode = false, bool $disableCookieCheck = false, bool $generateWarnings = false): ?array

    {
        if (is_null($this->messageParameters)) {
            $this->parseMessage($strictMode, $disableCookieCheck, $generateWarnings);
// Set debug mode
            if (!Util::$logLevel->logDebug()) {
                $this->debugMode = (isset($this->messageParameters['custom_debug']) &&
                    (strtolower($this->messageParameters['custom_debug']) === 'true'));
                if ($this->debugMode) {
                    Util::$logLevel = LogLevel::Debug;
                }
            }
// Set return URL if available
            if (!empty($this->messageParameters['lti_message_type']) &&
                (($this->messageParameters['lti_message_type'] === 'ContentItemSelectionRequest') || ($this->messageParameters['lti_message_type'] === 'ContentItemUpdateRequest')) &&
                !empty($this->messageParameters['content_item_return_url'])) {
                $this->returnUrl = $this->messageParameters['content_item_return_url'];
            }
            if (empty($this->returnUrl) && !empty($this->messageParameters['launch_presentation_return_url'])) {
                $this->returnUrl = $this->messageParameters['launch_presentation_return_url'];
            }
        }

        return $this->messageParameters;
    }
```

But when extra validation was added and the logic reordered, I think you meant to put !is_null on the secondary check instead of is_null.  Because of the is_null check, the returnUrl is never set.

In other words, the logic became:

1. Check strict mode
2. If ok, parse parameters if not set
3. If ok and parameters not set, set return URL. (this check fails because the parameters parsed correctly)

if ($this->ok && is_null($this->messageParameters)) {
            $this->parseMessage($disableCookieCheck, $generateWarnings);
        }
        if ($this->ok && is_null($this->messageParameters)) { **<-- This should change to !is_null right?**

**Comment by spvickers** [comment]
Thanks for reporting this issue; my apologies for creating it.  I will commit a fix to resolve it.
